src/interface/utils/Downloader.ts

```diff
@@ -0,0 +1,9 @@
+import { saveAs } from 'file-saver';
```

Comment:
We should avoid using the smaller kind of dependencies like this, and use the HTML5 API for this cases instead. FileSaver.js is maintained by just two people, and the last release published was two years ago.

I just generated similar functionality with ChatGPT, please try to adapt it here!

```
/**
 * Downloads a text file with the given filename and content.
 * @param filename The name of the file to be downloaded.
 * @param text The content of the text file.
 */
export const downloadTextFile = (filename: string, text: string): void => {
    const blob = new Blob([text], { type: 'text/plain;charset=utf-8' });
    downloadBlobFile(filename, blob);
};

/**
 * Downloads a JSON file with the given filename and object data.
 * @param filename The name of the file to be downloaded.
 * @param object The JSON object to be saved in the file.
 */
export const downloadJsonFile = (filename: string, object: Record<string, unknown>): void => {
    const text = JSON.stringify(object);
    const blob = new Blob([text], { type: 'application/json;charset=utf-8' });
    downloadBlobFile(filename, blob);
};

/**
 * Creates a download link for a Blob object and triggers the download.
 * @param filename The name of the file to be downloaded.
 * @param blob The Blob object to be saved as a file.
 */
const downloadBlobFile = (filename: string, blob: Blob): void => {
    const url = URL.createObjectURL(blob);
    const a = document.createElement('a');
    a.href = url;
    a.download = filename;
    a.style.display = 'none';
    document.body.appendChild(a);
    a.click();
    document.body.removeChild(a);
    URL.revokeObjectURL(url);
};
```

Reply:
Okay, fixing in the next commit.